app/controllers/admin/deploy_groups_controller.rb

```diff
@@ -91,7 +100,7 @@ def new_stage_with_group(stage)
     stage = Stage.build_clone(stage)
     stage.deploy_groups << deploy_group
     stage.name = deploy_group.name
-    stage.name << " -- copy #{SecureRandom.hex(4)}" if stage.project.stages.where(name: stage.name).exists?
```

Comment:
Remove the extra name bit, because we won't be creating duplicate stages for the same project/environment.